docs/reference/analysis/tokenfilters/keyword-marker-tokenfilter.asciidoc

```diff
@@ -1,137 +1,189 @@
 [[analysis-keyword-marker-tokenfilter]]
-=== Keyword Marker Token Filter
+=== Keyword marker token filter
+++++
+<titleabbrev>Keyword marker</titleabbrev>
+++++
 
-Protects words from being modified by stemmers. Must be placed before
-any stemming filters.
+Excludes specified tokens, called <<stemmer-keyword,keywords>>, from
+<<stemming,stemming>>.
 
-[cols="<,<",options="header",]
-|=======================================================================
-|Setting |Description
-|`keywords` |A list of words to use.
+The `keyword_marker` filter is often used with custom
+<<analysis-lang-analyzer,language analyzers>> to protect specific words from
+stemming. This filter uses Lucene's
+https://lucene.apache.org/core/{lucene_version_path}/analyzers-common/org/apache/lucene/analysis/miscellaneous/KeywordMarkerFilter.html[KeywordMarkerFilter].
 
-|`keywords_path` |A path (either relative to `config` location, or
-absolute) to a list of words.
+[TIP]
+====
+If you want to preserve both a stemmed and unstemmed version of each token,
+we recommend using the <<analysis-keyword-repeat-tokenfilter,`keyword_repeat`>>
+token filter.
```

Comment:
I don't think the "we recommend" is necessary--use the imperative:

To preserve both a stemmed and unstemmed version of each token, use the keyword_repeat token filter.

Reply:
Changed to imperative with 81244f6d8c78c39c4896463b7e03b5a26a7e71cd.

Updated the tip with 08de8710ffd.